### PR TITLE
Refactor direct funding to use AdvanceChannel

### PR DESCRIPTION
**Pull request by andrewgordstewart**

### CODE REVIEW

File: packages/wallet/src/redux/protocols/direct-funding/__tests__/scenarios.ts [hunk 1]
```diff
@@ -73,78 +51,68 @@ const bFundingReceivedEvent = globalActions.fundingReceivedEvent({
...
-const postFundSetup0 = globalActions.commitmentReceived({
-  processId: channelId,
-  signedCommitment: scenarios.signedCommitment2,
-});
-const postFundSetup1 = globalActions.commitmentReceived({
-  processId: channelId,
-  signedCommitment: scenarios.signedCommitment3,
-});
+
+const sharedData = () => ({ ...advanceChannelScenarios.initialStore });
```
Comment (lalexgap) [inline packages/wallet/src/redux/protocols/direct-funding/__tests__/scenarios.ts]: Is there a reason we need this function?

File: packages/wallet/src/redux/protocols/direct-funding/__tests__/scenarios.ts [hunk 2]
```diff
@@ -73,78 +51,68 @@ const bFundingReceivedEvent = globalActions.fundingReceivedEvent({
...
   },
   waitForPostFundSetup: {
     state: states.waitForFundingAndPostFundSetup({
       ...defaultsForA,
       channelFunded: true,
-      postFundSetupReceived: false,
     }),
-    sharedData: aHasPostFund0SharedData,
-    action: postFundSetup1,
+    sharedData: { ...advanceChannelScenarios.initialStore },
```
Comment (lalexgap) [inline packages/wallet/src/redux/protocols/direct-funding/__tests__/scenarios.ts]: Should this be `sharedData()` to be consistent?

File: packages/wallet/src/redux/protocols/direct-funding/__tests__/scenarios.ts (hunk 1 above)
Comment (andrewgordstewart) [reply packages/wallet/src/redux/protocols/direct-funding/__tests__/scenarios.ts]: I'm not sure. We re-assign to `sharedData` at many points in reducers, so the value of `advanceChannelScenarios.initialStore` may change throughout a test suite.

File: packages/wallet/src/redux/protocols/direct-funding/__tests__/scenarios.ts (hunk 2 above)
Comment (andrewgordstewart) [inline packages/wallet/src/redux/protocols/direct-funding/__tests__/scenarios.ts]:
```suggestion
    sharedData: sharedData(),
```

**Pull request merged**